File: internal/content/testdata/bench/java/is-factorion.java
import java.util.*;

class Solution {
    public boolean is_factorion(long n) {
        long original = n;
        long s = 0;
        while (n > 0) {
            long d = n % 10;
            s += factorial(d);
            n /= 10;
        }
        return s == original;
    }

    private long factorial(long x) {
        long f = 1;
        for (long i = 2; i <= x; i++) f *= i;
        return f;
    }
}

File: internal/content/testdata/bench/go/is-factorion.go
func is_factorion(n int) bool {
	factorial := func(x int) int {
		result := 1
		for i := 2; i <= x; i++ {
			result *= i
		}
		return result
	}
	original := n
	s := 0
	for n > 0 {
		d := n % 10
		s += factorial(d)
		n /= 10
	}
	return s == original
}
